### ISSUE TITLE
Unable to authenticate the request due to an error: Expected HTTP response code [204 404] when accessing [HEAD https://xxx.xxx.com:35357/v3/auth/tokens], but got 200 instead

### ISSUE THREAD

**Issue by dixudx** [gophercloud/gophercloud#202]
There is a bug in current v3 token validation.

> Unable to authenticate the request due to an error: Expected HTTP response code [204 404] when accessing [HEAD https://xxx.xxx.com:35357/v3/auth/tokens], but got 200 instead

Actually the valid return codes are,

```
Normal response codes: 200

Error response codes: 413,405,404,403,401,400,503
```

For more detailed information, refer to [keystone v3 token validate doc](http://developer.openstack.org/api-ref/identity/v3/index.html?expanded=validate-and-show-information-for-token-detail)

**Comment by jtopjian** [comment]
I can confirm this behavior, though I think the correct codes for this should be: 200, 400, 401, 403, 404. While the others are valid codes in Keystone, I'm not sure if they make sense in this context.

Here's a list of the Keystone exception codes: https://github.com/openstack/keystone/blob/master/keystone/exception.py

And here's prior research on the 200 response: https://github.com/gophercloud/gophercloud/issues/39

**Comment by jtopjian** [comment]
Fixed by #203
